src/main/java/core/basesyntax/model/FruitTransaction.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax.model;
+
+public class FruitTransaction {
+    private final Operation operation;
+    private final String fruit;
+    private final int quantity;
+
+    public FruitTransaction(Operation operation, String fruit, int quantity) {
+        this.operation = operation;
+        this.fruit = fruit;
+        this.quantity = quantity;
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public String getFruit() {
+        return fruit;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private final String operation;
+
+        Operation(String code) {
```

Comment:
The constructor parameter 'code' for the enum `Operation` could be renamed to something more descriptive like 'operationCode' to improve code readability.